### PR TITLE
Add language guide for Javascript

### PR DESCRIPTION
**Pull request by glokta1**

### CODE REVIEW

File: docs/challenges/language-support/javascript.mdx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+For example, if you add a file at `app/foo.js`, you could use it like so:
+
+<CodeGroup>
+```javascript app/foo.js
+function foo() {
+    console.log('Hello from foo!');
+}
+
+function bar() {
+console.log('Hello from bar!');
```
Comment (rohitpaulk) [inline docs/challenges/language-support/javascript.mdx:37]: Indentation

File: docs/challenges/language-support/javascript.mdx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+module.exports = { foo, bar };
+
```
Comment (rohitpaulk) [inline docs/challenges/language-support/javascript.mdx:41]: Extra newline

File: docs/challenges/language-support/javascript.mdx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+````
+
+```javascript app/main.js
+const { foo, bar } = require('./foo');
+
+function main() {
+    foo();  // prints 'Hello from foo!'
```
Comment (rohitpaulk) [inline docs/challenges/language-support/javascript.mdx:48]:
```suggestion
    foo();  // Uses the foo function from foo.js
```

(Consistent with what we do for other languages + this is the more important part to highlight)

File: docs/challenges/language-support/javascript.mdx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    bar();  // prints 'Hello from bar!'
+}
+````
+
+</CodeGroup>
+
+<Note>We use the CommonJS syntax for `import` / `export` of modules.</Note>
```
Comment (rohitpaulk) [inline docs/challenges/language-support/javascript.mdx:55]: @glokta1 can you clarify what this means? We don't seem to be using the `import` keyword no?

File: docs/challenges/language-support/javascript.mdx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+# Adding dependencies
+
+You can add dependencies to your project using `npm`
```
Comment (rohitpaulk) [inline docs/challenges/language-support/javascript.mdx:59]: @glokta1 Let's add a note here saying we haven't added instructions yet. Once we've upgraded Node I'll get Ryan to share details here

File: docs/challenges/language-support/javascript.mdx (hunk 4 above)
Comment (glokta1) [reply docs/challenges/language-support/javascript.mdx:55]: @rohitpaulk I think surrounding import and export with the code blocks is making it unclear. I simply meant we're using CommonJS for importing and exporting of files. I'll change this.

**Review by rohitpaulk — APPROVED** [review]

**Pull request merged**